### DISCUSSION TITLE
1 opcua session vs multiple sessions

### DISCUSSION THREAD

**Discussion by jordanjj8** [FreeOpcUa/python-opcua#1383] [Q&A] [answered]
Hey there, I have a program that runs 4 threads where each thread gets/sets multiple node values constantly. I am finding that I am getting PLC timeout issues for that specific program, but the odd thing is that I also have other programs running in different sessions but Im not seeingg any PLC timeout issues. This makes me think I am overloading the session. I was wondering if using one session for each thread instead of using one session for handling all 4 threads is recommended.

Also in general, is it better to use less sessions or more sessions? Or is it a balance between number of sessions and how hard one session is working?

**Comment by oroulet** [comment, marked as answer]
Code should be thread safe. So in most cases using several threads and one session should be better. But if you are doing a lot of computation maybe you need several processes. Really depends on you use case
